client/db/src/bench.rs

```diff
@@ -121,10 +223,12 @@ impl<B: BlockT> StateBackend<HashFor<B>> for BenchmarkingState<B> {
 	type TrieBackendStorage = <DbState<B> as StateBackend<HashFor<B>>>::TrieBackendStorage;
 
 	fn storage(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
+		self.add_read_key(key);
```

Comment:
Did you see some significant change in bench performance from this? (it adds some map query/update so I am curious about the impact).

Reply:
@cheme There is a negligible difference in timing that I can see across two modules:

```
diff --git a/summary.json b/summary.json
index 5d8f507..6742744 100644
--- a/summary.json
+++ b/summary.json
@@ -1,26 +1,26 @@
 {
 	"balances_set_balance": {
-		"base": "27.56",
+		"base": "27.19",
 		"u": "0",
 		"e": "0"
 	},
 	"balances_set_balance_killing": {
-		"base": "35.11",
+		"base": "33.59",
 		"u": "0",
 		"e": "0"
 	},
 	"balances_transfer": {
-		"base": "73.63",
+		"base": "65.23",
 		"u": "0",
 		"e": "0"
 	},
 	"balances_transfer_best_case": {
-		"base": "39.06",
+		"base": "34.88",
 		"u": "0",
 		"e": "0"
 	},
 	"balances_transfer_keep_alive": {
-		"base": "51.4",
+		"base": "46.17",
 		"u": "0",
 		"e": "0"
 	},
@@ -61,50 +61,54 @@
 		"c": "0"
 	},
 	"collective_close_approved": {
-		"base": "65.95",
+		"base": "68.62",
 		"m": "0.226",
-		"p": "0.487",
-		"b": "0.005"
+		"p": "0.447",
+		"b": "0.004"
 	},
 	"collective_close_disapproved": {
-		"base": "51.08",
-		"m": "0.224",
-		"p": "0.475",
-		"b": "0.003"
+		"base": "45.99",
+		"m": "0.259",
+		"p": "0.447",
+		"b": "0"
 	},
 	"collective_close_early_approved": {
-		"base": "50.82",
-		"m": "0.211",
-		"p": "0.478",
-		"b": "0.008"
+		"base": "63.57",
+		"m": "0.243",
+		"p": "0.423",
+		"b": "0.002"
 	},
 	"collective_close_early_disapproved": {
-		"base": "37.21",
-		"m": "0.239",
-		"p": "0.466",
-		"b": "0.002"
+		"base": "38.98",
+		"m": "0.24",
+		"p": "0.461",
+		"b": "0.003"
 	},
 	"collective_execute": {
-		"base": "22.62",
-		"m": "0.115",
-		"b": "0.003"
+		"base": "23.09",
+		"m": "0.112",
+		"b": "0.004"
 	},
 	"collective_propose_execute": {
-		"base": "28.12",
+		"base": "28.26",
 		"m": "0.218",
-		"b": "0.003"
+		"b": "0.004"
 	},
 	"collective_propose_proposed": {
-		"base": "49.75",
-		"m": "0.105",
-		"p": "0.502",
-		"b": "0.006"
+		"base": "53.66",
+		"m": "0.111",
+		"p": "0.427",
+		"b": "0.002"
 	},
 	"collective_set_members": {
 		"base": "0",
-		"m": "20.47",
-		"n": "0.109",
-		"p": "26.29"
+		"m": "20.51",
+		"n": "0.136",
+		"p": "25.72"
+	},
+	"collective_vote": {
+		"base": "37.39",
+		"m": "0.302"
 	},
 	"collective_vote_insert": {
 		"base": "24.03",
```

Also our main DB benchmarks are done in a separate test using the real DB backend.

Reply:
So these are on in memorydb, thanks for posting me these it interesting numbers.